packages/contracts/contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol

```diff
@@ -69,6 +107,19 @@ contract OVM_ECDSAContractAccount is iOVM_ECDSAContractAccount {
             Lib_ExecutionManagerWrapper.ovmCHAINID()
         );
 
+        // Compute the intrinsic gas
+        uint256 intrinsicGas = _intrinsicGas(
+            _encodedTransaction,
+            transaction.isCreate
+        );
+
+        // Need to make sure that the gas is sufficient to execute the transaction.
```

Comment:
Does this need to be `gasleft() - intrinsicGas >= gasLimit` ?

Reply:
Oh, good point.  Maybe the correct way to do it might be have `uint initialGas = gasleft()` at the very first line of `execute`, and then here check that `initialGas - intrinsicGas >= gasLimit`? That way the decoding and intrinsic gas calculation can be considered "a part of" the intrinsic gas."